main/streams/streams.c

```diff
@@ -1124,6 +1135,8 @@ PHPAPI zend_string *php_stream_get_record(php_stream *stream, size_t maxlen, con
 static ssize_t _php_stream_write_buffer(php_stream *stream, const char *buf, size_t count)
 {
 	ssize_t didwrite = 0;
+	ssize_t retval;
+	bool old_eof = stream->eof;
```

Comment:
I think seek should not change it in this case but worth to double check it as seek set resets eof in some cases for some streams. Might be possibly just safer to take this value after the seek.